packages/vite/src/devFeServer.ts

```diff
@@ -13,11 +13,11 @@ import {
   createPerRequestMap,
   createServerStorage,
 } from '@redwoodjs/server-store'
+import type { Middleware } from '@redwoodjs/web/dist/server/middleware.js'
```

Comment:
This looks strange. Shouldn't we import types from `src/`?
Or even better, if we can, like this `import type { Middleware } from '@redwoodjs/web/middleware'`

Reply:
The vite package isn't ESM or moduleRes next :)

Reply:
Never tried the src import though. Can take a look!

Reply:
So took a look at this, technically types we COULD import from src, but cases where we need the actual class that's not correct (it should import from dist, until the vite package is converted to moduleRes: next). I want to keep that in a separate PR though.

Reply:
Honestly I'm surprised this even works, because when it *isn't* a class, is there even a type in dist to import? Or is the tooling smart enough to actually look at the `.d.ts` file next to the file you're pointing to?

Reply:
Types are _always_ in dist - that's the point of types in a library no? Only when you are importing internally within a package do you import the type from source. In this case, the web package is a library to the vite package, as they are separate packages.

If you import like this `import x from 'bazinga'` it will look at bazinga/dist/index.d.ts - this is the most common approach.

Reply:
Yeah, I guess I just got confused by the fact that you're specifying the specific file, extension and all in this case. 
(It's also still a little bit weird to me that you have to specify `.js` when importing ts files. And yes, I know *why* that's the case, but still looks weird 😀)

Reply:
All good though. Keep it as dist imports 👍